### PR TITLE
Allow NotifcationManager to be created on demand

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
Make `NotificationManager` attach to the Toplevel on creation, instead of on template applied on the host. This allows managed notifications to be created through the TopLevel on all platforms.

## What is the current behavior?
Managed notifications are only created using a `WindowNotificationManager` instance that requires a `Window` object. This prevents single view platforms from accessing it.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

File: src/Avalonia.Controls/TopLevel.cs [hunk 1]
```diff
@@ -328,6 +331,8 @@ bool IInputRoot.ShowAccessKeys
 
         IRenderTarget IRenderRoot.CreateRenderTarget() => CreateRenderTarget();
 
+        public WindowNotificationManager NotificationManager => _notificationManager;
```
Comment (maxkatz6) [inline src/Avalonia.Controls/TopLevel.cs:334]: Let's make initialization of NotificationManager lazy. I.e., create object (and attach) only after getter was called for the first time.
As I see from the tests, current implementation also affects popups, which has a top level too iirc. Having this initialized for all of them even if not needed is a bit unnecessary.

Other than that looks good.

Comment (maxkatz6) [inline src/Avalonia.Controls/TopLevel.cs:334]: Let's make initialization of NotificationManager lazy. I.e., create object (and attach) only after getter was called for the first time.
As I see from the tests, current implementation also affects popups, which has a top level too iirc. Having this initialized for all of them even if not needed is a bit unnecessary.

Other than that looks good.

**Review by maxkatz6 — CHANGES_REQUESTED** [review]

File: src/Avalonia.Controls/TopLevel.cs (hunk 1 above)
Comment (emmauss) [reply src/Avalonia.Controls/TopLevel.cs]: Done.

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026048-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026123-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**